src/test/java/com/uid2/admin/vertx/test/ServiceTestBase.java

```diff
@@ -147,16 +148,40 @@ protected void get(Vertx vertx, String endpoint, Handler<AsyncResult<HttpRespons
         client.getAbs(getUrlForEndpoint(endpoint)).send(handler);
     }
 
+    protected void get(Vertx vertx, VertxTestContext testContext, String endpoint, Handler<AsyncResult<HttpResponse<Buffer>>> handler) {
+        val wrappedHandler = wrapHandlerToCatchAssertionErrors(testContext, handler);
+        get(vertx, endpoint, wrappedHandler);
+    }
+
     protected void post(Vertx vertx, String endpoint, String body, Handler<AsyncResult<HttpResponse<Buffer>>> handler) {
         WebClient client = WebClient.create(vertx);
         client.postAbs(getUrlForEndpoint(endpoint)).sendBuffer(Buffer.buffer(body), handler);
     }
 
+    protected void post(Vertx vertx, VertxTestContext testContext, String endpoint, String body, Handler<AsyncResult<HttpResponse<Buffer>>> handler) {
+        val wrappedHandler = wrapHandlerToCatchAssertionErrors(testContext, handler);
+        post(vertx, endpoint, body, wrappedHandler);
+    }
+
     protected void postWithoutBody(Vertx vertx, String endpoint, Handler<AsyncResult<HttpResponse<Buffer>>> handler) {
         WebClient client = WebClient.create(vertx);
         client.postAbs(getUrlForEndpoint(endpoint)).sendBuffer(null, handler);
     }
 
+    protected void postWithoutBody(Vertx vertx, VertxTestContext testContext, String endpoint, Handler<AsyncResult<HttpResponse<Buffer>>> handler) {
+        val wrappedHandler = wrapHandlerToCatchAssertionErrors(testContext, handler);
+        postWithoutBody(vertx, endpoint, wrappedHandler);
+    }
+
+    protected static Handler<AsyncResult<HttpResponse<Buffer>>> wrapHandlerToCatchAssertionErrors(VertxTestContext testContext, Handler<AsyncResult<HttpResponse<Buffer>>> handler) {
+        return result -> {
```

Comment:
are you sure there's not a library method which does this?

Reply:
https://github.com/eclipse-vertx/vertx-junit5/blob/6205224f5168042f99fff1bb5072828c0f21e69a/src/main/java/io/vertx/junit5/VertxTestContext.java#L373-L381 ?

Reply:
https://vertx.io/docs/vertx-junit5/java/#_a_test_context_for_asynchronous_executions

based on the documentation here. I think wrap them with a call to 
verify, that succeeding or failing is the right way to do it. However, most of our tests are making assertions on asyncResult rather than the response. I found wrap with failNow make less changes to each test